### PR TITLE
Fix triangle position

### PR DESCRIPTION
**Pull request by marcoacfilho**
**Release Type:** *Breaking Changes* 

Fixes https://x-team-internal.atlassian.net/browse/XP-2946

## Description

Created story to better test the ListExclusionFormPopup in a real use case, and fixed arrow positioning.

## Checklist

- [x] set yourself as an assignee
- [x] set appropriate labels for a PR (`In Review` or `In Progress` depending on its status)
- [x] move respective JIRA issue to the `IN REVIEW` column
- [x] make sure your code lints (`npm run lint`)
- [x] Flow typechecks passed (`npm run flow`)
- [x] Snapshots tests passed (`npm run jest`)
- [x] component's documentation (`.stories.js` file) is changed or added accordingly to reflect any new or updated use cases or variants usage
- [x] if any snapshots have been changed, verify that component still works and looks as expected and update the changed snapshot
- [x] **manually tested the app** by running it in several different browsers (Firefox, Chrome, Opera, Safari, MS IE/Edge, etc.) and checked nothing is broken and operates as expected!

## Related PRs

xp_ui | [XP](https://github.com/x-team/xp-ui/pull/374)

## Steps to Test or Reproduce

1. Search for ListExclusionFormPopup 
2. Test the different use cases 

## Impacted Areas in Application

- `ui/ListExclusionFormPopup.js`
- `ui/ListExclusionFormPopup.stories.js`

## Screenshots

image

image

### CODE REVIEW

File: src/components/ui/ListExclusionFormPopup.stories.js [hunk 1]
```diff
@@ -37,45 +39,118 @@ const StoryListExclusionFormPopup = (props) => (
...
+  )
+}
+storiesOf('UI Components/ListExclusionFormPopup', module).add('standalone', () => (
+  <Body>
+    <StoryListExclusionFormPopup />
+  </Body>
+))
 
 storiesOf('UI Components/ListExclusionFormPopup/Debug', module)
   .add('stateful example composed with TwoColumnsLayout', () => (
```
Comment (bernardodiasc) [inline src/components/ui/ListExclusionFormPopup.stories.js]: I guess this story can be moved to `UI Components/ListExclusionFormPopup/Use Cases` category.

File: src/components/ui/ListExclusionFormPopup.stories.js [hunk 2]
```diff
@@ -37,45 +39,118 @@ const StoryListExclusionFormPopup = (props) => (
...
 const ScrollableContainer = ({ scrollable = true, children }) => (
-  <div style={{
-    height: '100%',
-    overflow: scrollable ? 'auto' : 'hidden'
-  }}>
+  <div
+    style={{
+      height: '100%',
+      overflow: scrollable ? 'auto' : 'hidden',
+      paddingRight: '30%',
```
[the remark this answers is not in the document]
Comment (marcoacfilho) [reply src/components/ui/ListExclusionFormPopup.stories.js]: It was used so the Popup wouldn't open in the outside right part of the screen creating a horizontal scroll. I can change to absolute.

File: src/components/ui/ListExclusionFormPopup.stories.js [hunk 3]
```diff
@@ -5,30 +5,32 @@ import { number, text, array } from '@storybook/addon-knobs'
...
+        height: 0,
+        top: 0,
+        right: 0,
+        bottom: 0,
+        left: 0
+      }
+    }
     marginTop={number('Margin Top', props.marginTop || 10)}
     marginBottom={number('Margin Bottom', props.marginBottom || 10)}
+    marginRight={number('Margin Right', props.marginRight || 10)}
```
[the remark this answers is not in the document]
Comment (marcoacfilho) [reply src/components/ui/ListExclusionFormPopup.stories.js]: Done

File: src/components/ui/ListExclusionFormPopup.stories.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcoacfilho) [reply src/components/ui/ListExclusionFormPopup.stories.js]: Changed to absolute

File: src/components/ui/ListExclusionFormPopup.stories.js [hunk 4]
```diff
@@ -112,12 +188,10 @@ storiesOf('UI Components/ListExclusionFormPopup/Debug', module)
             />
           )}
           <ScrollableContainer scrollable={!state.isOpen}>
-            <ButtonsList setState={setState} />
+            <ApplicantList setState={setState} />
```
[the remark this answers is not in the document]
Comment (marcoacfilho) [reply src/components/ui/ListExclusionFormPopup.stories.js]: Done

File: src/components/ui/ListExclusionFormPopup.stories.js (hunk 1 above)
Comment (marcoacfilho) [reply src/components/ui/ListExclusionFormPopup.stories.js]: Done

File: src/components/ui/ListExclusionFormPopup.js [hunk 5]
```diff
@@ -241,6 +241,14 @@ class ListExclusionFormPopup extends PureComponent<Props, State> {
       style.top = marginTop
       anchor = this.getAnchorPositionForScreenSmallerThanPopupMaxHeight({ bottom, innerHeight, marginBottom, margins, triangleHeight, top, marginTop, initialAnchor })
     } else {
+      if (height < triangleHeight) {
```
[the remark this answers is not in the document]
Comment (marcoacfilho) [reply src/components/ui/ListExclusionFormPopup.js]: Fixed

File: src/components/ui/ListExclusionFormPopup.js [hunk 6]
```diff
@@ -241,6 +241,14 @@ class ListExclusionFormPopup extends PureComponent<Props, State> {
...
+        if (this.isClickedElementTopAboveThePopupTop({ top: top - triangleHeight, marginTop })) {
+          style.top = marginTop
+        } else {
+          style.top = top - triangleHeight
```
[the remark this answers is not in the document]
Comment (marcoacfilho) [reply src/components/ui/ListExclusionFormPopup.js]: Done

File: src/components/ui/ListExclusionFormPopup.js [hunk 7]
```diff
@@ -241,6 +241,14 @@ class ListExclusionFormPopup extends PureComponent<Props, State> {
...
+        }
+        anchor = triangleHeight - (height / 2)
```
[the remark this answers is not in the document]
Comment (marcoacfilho) [reply src/components/ui/ListExclusionFormPopup.js]: Done

File: src/components/ui/ListExclusionFormPopup.js [hunk 8]
```diff
@@ -234,13 +238,21 @@ class ListExclusionFormPopup extends PureComponent<Props, State> {
       height: maxHeight,
       maxHeight: `calc(100vh - ${margins}px)`,
       top,
-      left: left + width + 12
+      left: left + width + 10
```
Comment (nicksp) [inline src/components/ui/ListExclusionFormPopup.js]: Could you tell me please where this "magic" number (`10`) comes from? Is there a way to define it as a variable with a meaningful name so it's easier to recall its purpose in the long run, after some time working with app?

File: src/components/ui/ListExclusionFormPopup.stories.js [hunk 9]
```diff
@@ -70,14 +78,83 @@ const ButtonsList = ({ setState }) => Array(50)
...
+            setState({
+              isOpen: true,
+              applicant: `Applicant full name`,
+              positioning: positioning
+            })
+        }
+      ]}
+    />
+  )
+  items[1] = (
```
Comment (nicksp) [inline src/components/ui/ListExclusionFormPopup.stories.js]: Sorry if I missed that, but why do we need to redefine `items[1]` here? is that a key part? I mean, we're duplicating the definition for `ApplicantBadge` here that we already defined above, so wondering if that was intentional or a leftover.

File: src/components/ui/ApplicantBadge.js [hunk 10]
```diff
@@ -675,22 +675,18 @@ class ApplicantBadge extends PureComponent<Props> {
           )}
         </div>
         <div className={cx.controls}>
-          {!isTabular && actions.map(({ key, icon: Icon, onClick = null, render, dropdownClassName, tooltipClassName }) => (
```
Comment (nicksp) [inline src/components/ui/ApplicantBadge.js]: @marcoacfilho Please check https://github.com/x-team/xp-ui/pull/374 for recent changes. Seems like we need to rebase with this branch, right? I think that what you did before. Please note that we did quite several changes to `ApplicantBadge.js` component and most likely we'd like to sync them up. Or are changes you introduce here for that file done intentionally and should override ones on https://github.com/x-team/xp-ui/pull/374 PR? If they just come from that PR, please rebase again to update them.

**Comment by nicksp** [comment]
Merging this one with an idea of addressing this review notes in the scope of #374 cause seems like these branches aren't in sync and might make sense to focus on these things in one scope to prevent diff.

File: src/components/ui/ListExclusionFormPopup.js (hunk 8 above)
Comment (marcoacfilho) [reply src/components/ui/ListExclusionFormPopup.js]: Done

File: src/components/ui/ListExclusionFormPopup.stories.js (hunk 9 above)
Comment (marcoacfilho) [reply src/components/ui/ListExclusionFormPopup.stories.js]: this is used to make the selected ApplicantBadge

**Pull request merged**